packages/web-app-preview/src/App.vue

```diff
@@ -543,6 +540,78 @@ export default defineComponent({
     imageRotateRight() {
       this.currentImageRotation =
         this.currentImageRotation === 270 ? 0 : this.currentImageRotation + 90
+    },
+    isFileTypeImage(file) {
+      return !this.isFileTypeAudio(file) && !this.isFileTypeVideo(file)
+    },
+    isFileTypeAudio(file) {
+      return file.mimeType.toLowerCase().startsWith('audio')
+    },
+
+    isFileTypeVideo(file) {
+      return file.mimeType.toLowerCase().startsWith('video')
+    },
+    preloadImages() {
+      let toPreloadFiles = []
+
+      const loadPreviewAsync = (file) => {
+        toPreloadFiles.push(file.id)
+        loadPreview({
+          resource: file,
+          isPublic: this.isPublicLinkContext,
+          server: configurationManager.serverUrl,
+          userId: this.user.id,
+          token: this.accessToken,
+          dimensions: [this.thumbDimensions, this.thumbDimensions] as [number, number]
+        })
+          .then((mediaUrl) => {
+            this.cachedFiles.push({
```

Comment:
This duplicates some code with the `loadActiveFileIntoCache` method (the `this.cachedFiles.push...` call). could you refactor that into a separate method?